eclipse/che
# Factories for private Gitlab repository with pre-created personal access token

### What does this PR do?
This PR allows to accept factories to private Gitlab repositories using pre-created personal access tokens stored as a secret inside the user namespace. Secret example is provided below:

```
kind: Secret
apiVersion: v1
metadata:
  annotations:
    che.eclipse.org/che-userid: 73a27d90-3d63-4d8c-a8c6-9b4a716c702a
    che.eclipse.org/expired-after: '-1'
    che.eclipse.org/scm-url: 'https://gitlab.apps.cluster-2ab2.2ab2.example.opentlc.com'
    che.eclipse.org/scm-userid: '1'
    che.eclipse.org/scm-username: root
  name: gitlab-personal-access-token-secret
  labels:
    app.kubernetes.io/component: scms-personal-access-token
    app.kubernetes.io/part-of: che.eclipse.org
data:
  token: Yzh5cEt6cURxUWVCa3FKazhtaHg=
type: Opaque
```

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19170

### How to test this PR?
Prepare private Gitblab repository;
Create and save personal access token on Gitlab side;
Configure Che server to use Gitlab instance and prepare secret in user namespace;
Run factory to prevate repo. Factory should work.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
